app/src/main/java/io/intrepid/contest/models/Contest.java

```diff
@@ -14,7 +14,7 @@
     private Date lastUpdatedDate;
     private String title;
     private String description;
-    private List<Category> categories;
+    private List<Category> categories = new ArrayList<>();
```

Comment:
For now, I'm not sure I really need a constructor for this class yet. I don't think there's a performance difference here. I just didn't want to define a constructor just for setting up the categories. Most likely, the constructor will be a restricted private one, and all of these fields will be provided by the builder. If I find myself initializing a bunch of things in the constructor, then that might be an indicator that the categories should be initialized in there. This is primarily to avoid an NPE. But i suspect in the final iteration, the categories List will just be passed in to the instance of contest.